app/services/user_authenticator.rb

```diff
@@ -20,8 +20,18 @@ def has_authenticator?
     !!authenticator
   end
 
+  def confirm_email
```

Comment:
I would prefer to keep this method private and test `UserAuthenticator#finish` instead.